### PR TITLE
[hotfix]Fix some logging statements which can not correctly record the behaviors

### PR DESCRIPTION
**Pull request by lzh3636**
## Documentation

  - Does this pull request introduce a new feature? (no) 

There are some possible copy and paste errors in the log messages (The logging statement was copied from an old place to a new place, but the message wasn't changed to adapt to the function of the new place).
So I modified some logging statements to make them record the run-time behaviors more accurately.

### CODE REVIEW

File: flink-runtime/src/main/java/org/apache/flink/runtime/jobmaster/JobMaster.java [hunk 1]
```diff
@@ -759,7 +759,7 @@ public void declineCheckpoint(DeclineCheckpoint decline) {
 
 				return CompletableFuture.completedFuture(Acknowledge.get());
 			} catch (Exception e) {
-				log.error("Failed to notify KvStateRegistry about registration {}.", registrationName);
+				log.error("Failed to notify KvStateRegistry about registration {}.", registrationName, e);
```
Comment (zentol) [inline flink-runtime/src/main/java/org/apache/flink/runtime/jobmaster/JobMaster.java]: @kl0u Are we intentionally not logging the exception?

File: flink-runtime/src/main/java/org/apache/flink/runtime/highavailability/nonha/leaderelection/SingleLeaderElectionService.java [hunk 2]
```diff
@@ -171,7 +171,7 @@ public boolean hasLeadership(@Nonnull UUID leaderSessionId) {
 	}
 
 	void errorOnGrantLeadership(LeaderContender contender, Throwable error) {
-		LOG.warn("Error notifying leader listener about new leader", error);
+		LOG.warn("Error granting leadership to contender", error);
```
Comment (zentol) [inline flink-runtime/src/main/java/org/apache/flink/runtime/highavailability/nonha/leaderelection/SingleLeaderElectionService.java]: @tillrohrmann WDYT?

File: flink-runtime/src/main/java/org/apache/flink/runtime/jobmaster/JobMaster.java (hunk 1 above)
Comment (zentol) [reply flink-runtime/src/main/java/org/apache/flink/runtime/jobmaster/JobMaster.java]: @kl0u ping

File: flink-runtime/src/main/java/org/apache/flink/runtime/highavailability/nonha/leaderelection/SingleLeaderElectionService.java (hunk 2 above)
Comment (zentol) [reply flink-runtime/src/main/java/org/apache/flink/runtime/highavailability/nonha/leaderelection/SingleLeaderElectionService.java]: @tillrohrmann ping

File: flink-runtime/src/main/java/org/apache/flink/runtime/jobmaster/JobMaster.java (hunk 1 above)
Comment (kl0u) [reply flink-runtime/src/main/java/org/apache/flink/runtime/jobmaster/JobMaster.java]: I think adding the exception logging is correct. It is logged at `error` level so it is the right thing to do I suppose.

File: flink-runtime/src/main/java/org/apache/flink/runtime/highavailability/nonha/leaderelection/SingleLeaderElectionService.java (hunk 2 above)
Comment (tillrohrmann) [reply flink-runtime/src/main/java/org/apache/flink/runtime/highavailability/nonha/leaderelection/SingleLeaderElectionService.java]: LGTM.

**Pull request merged**